wp_api/src/reqwest_request_executor.rs

```diff
@@ -143,3 +159,227 @@ impl RequestExecutor for ReqwestRequestExecutor {
+                redirects: None,
+                reason: hyper_error.into(),
+            };
+        }
+
+        if let Some(dns_error) = error.as_dns_error() {
+            return RequestExecutionError::RequestExecutionFailed {
+                status_code: None,
+                redirects: None,
+                reason: dns_error.into(),
+            };
+        }
+
+        RequestExecutionError::RequestExecutionFailed {
+            status_code: None,
+            redirects: None,
+            reason: RequestExecutionErrorReason::GenericError {
+                error_message: error.to_string(),
+            },
+        }
+    }
+}
+
+/// Converts all HTTPS errors to a RequestExecutionErrorReason
+impl From<&TlsError> for RequestExecutionErrorReason {
+    fn from(error: &TlsError) -> Self {
+        match error {
+            TlsError::InvalidCertificate(CertificateError::NotValidForNameContext {
+                expected,
+                presented,
+            }) => RequestExecutionErrorReason::InvalidSslError {
+                inner: InvalidSslError::CertificateNotValidForName {
+                    hostname: expected.to_str().to_string(),
+                    presented_hostnames: presented.to_vec(),
+                },
+            },
+            _ => RequestExecutionErrorReason::GenericError {
+                error_message: error.to_string(),
+            },
+        }
+    }
+}
+
+/// Converts all DNS errors to a RequestExecutionErrorReason
+impl From<&ResolveError> for RequestExecutionErrorReason {
+    fn from(error: &ResolveError) -> Self {
+        // Future improvement: We could probably detect when the domain is valid, but
+        // there's no DNS record for the provided hostname
+
+        RequestExecutionErrorReason::NonExistentSiteError {
+            error_message: Some(error.to_string()),
+            suggested_action: None,
+        }
+    }
+}
+
+/// Converts from the Hyper frameworks underlying errors to a RequestExecutionErrorReason
+impl From<&HyperError> for RequestExecutionErrorReason {
+    fn from(error: &HyperError) -> Self {
+        if let Some(http2_error) = error.find::<Http2Error>() {
+            // TODO: We can probably handle more cases here, such as:
+            // - Connection reset
+
+            return RequestExecutionErrorReason::GenericError {
+                error_message: http2_error.to_string(),
+            };
+        }
+
+        if error.is_closed() {
+            return RequestExecutionErrorReason::GenericError {
+                error_message: error.to_string(),
+            };
+        }
+
+        if error.is_incomplete_message() {
+            // The server terminated the connection unexpectedly
+            return RequestExecutionErrorReason::GenericError {
+                error_message: error.to_string(),
+            };
+        }
+
+        RequestExecutionErrorReason::GenericError {
+            error_message: error.to_string(),
+        }
```

Comment:
Replaced the `todo!("Unhandled hyper error: {:?}", error);` from the original implementation since we can't have `todo!()`s in production code.